### DISCUSSION TITLE
Best practices to handle two types of user identity?

### DISCUSSION THREAD

**Discussion by winsoroaks** [vercel/next.js#60362] [App Router]
Hi team, imagine im trying to build a two-sided marketplace, something like a a simplified Uber Eats for my web app. A restaurant owner can also simultaneously be a customer of other restaurants by logging into their account using the same email address. 

How should I design my routes such that I allow users to "select" if they are logging in as a restaurant owner or a customer? I'd like the home page to display different stuff from logging in as a restaurant owner. For example, as a customer, you should see what are your past orders at the homepage but as a restaurant owner, you should see how much sales you did at the homepage.

Should I be doing something like customers.MYWEBSITE.com vs owner.MYWEBSITE.com? Is there a prior art for this?

Thank you!

**Comment by ananuness** [comment]
Hmmm, have you thought about using roles for your users? And manage your routes based in which role they're assigned from next middlewares? You can check this [example](https://github.com/vercel/next.js/discussions/59884#discussioncomment-7930922) and see if it would help you!

**Reply by winsoroaks** [reply]
thank you. yes, i have. but if someone has both customer and business owner role, how can i separate their paths? 😅
